trusted-types/block-text-node-insertion-into-svg-script-element.html

```diff
@@ -145,47 +146,52 @@
   }, "Prep for subsequent tests: Create default policy.");
 
   for (let [element, create_element] of Object.entries(script_elements)) {
-    promise_test(t => {
+    promise_test(async t => {
       let s = create_element(document);
       let text = document.createTextNode("postMessage('default', '*');");
       s.appendChild(text);
-      container.appendChild(s);
-
-      return Promise.all([checkMessage(1),
-                            checkSecurityPolicyViolationEvent(0)]);
+      await no_trusted_type_violation_for(async _ =>
+        await checkMessage(_ => container.appendChild(s), 1)
+      );
     }, "Test that default policy applies. " + element);
 
-    promise_test(t => {
+    promise_test(async t => {
       let s = create_element(document);
       let text = document.createTextNode("fail");
       s.appendChild(text);
-      container.appendChild(s);
-      return Promise.all([checkMessage(0),
-                          checkSecurityPolicyViolationEvent(1)]);
+      await trusted_type_violation_without_exception_for(async _ =>
+        await checkMessage(_ => container.appendChild(s), 0)
+      );
     }, "Test a failing default policy. " + element);
   }
 
-  promise_test(t => {
+  promise_test(async t => {
     const inserted_script = document.getElementById("script1");
```

Comment:
This subtest is checking HTMLScriptElement when it should be testing SVGScriptElement.

Reply:
Okay scratch that I've just realised https://github.com/web-platform-tests/wpt/pull/50861 fixes that. (Maybe we should just squash them into one PR? I'll leave you to decide on that, I do want the default policy assertion not working fully fixed though (but if it's a wider issue feel free to move it to its own issue and we can address it througouut)